src/client/mesh_generator_thread.cpp

```diff
@@ -249,6 +244,14 @@ void MeshUpdateWorkerThread::doUpdate()
 		m_manager->putResult(r);
 		m_queue_in->done(q->p);
 		delete q;
+		sp.stop();
+
+		porting::TriggerMemoryTrim();
+
+		// do this after we're done so the interval is enforced without
+		// adding extra latency.
+		if (m_generation_interval)
+			sleep_ms(m_generation_interval);
```

Comment:
This does not change anything.

Did you mean to subtract the time the loop iteration took?

Reply:
Yes it does. Imagine the code blocking on an empty queue.
Before: request is taken, then 10ms of wait time, then it's processed.
After: request instantly processed, then 10ms before considering the next one.

it doesn't change anything *if* the queue never runs empty.